hudi-spark/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -52,29 +52,18 @@
  */
 public class DataSourceUtils {
 
-  /**
-   * Obtain value of the provided nullable field as string, denoted by dot notation. e.g: a.b.c
-   */
-  public static String getNullableNestedFieldValAsString(GenericRecord record, String fieldName) {
-    try {
-      return getNestedFieldValAsString(record, fieldName);
-    } catch (HoodieException e) {
-      return null;
-    }
-  }
-
   /**
    * Obtain value of the provided field as string, denoted by dot notation. e.g: a.b.c
    */
-  public static String getNestedFieldValAsString(GenericRecord record, String fieldName) {
-    Object obj = getNestedFieldVal(record, fieldName);
-    return obj.toString();
+  public static String getNestedFieldValAsString(GenericRecord record, String fieldName, boolean returnNullValue) {
+    Object obj = getNestedFieldVal(record, fieldName, returnNullValue);
+    return (obj == null) ? null : obj.toString();
   }
 
   /**
    * Obtain value of the provided field, denoted by dot notation. e.g: a.b.c
    */
-  public static Object getNestedFieldVal(GenericRecord record, String fieldName) {
+  public static Object getNestedFieldVal(GenericRecord record, String fieldName, boolean returnNullValue) {
```

Comment:
Can you rename to returnNullIfNotFound ?

Reply:
Done.